app/controllers/areas_controller.rb

```diff
@@ -18,7 +18,11 @@ def index
   end
 
   def search
-    api_response = Imminence.mapit_api.location_for_postcode(params[:postcode])
+    # Strip trailing whitespace, most non-alphanumerics, and use the
+    # uk_postcode gem to potentially transpose O/0 and I/1.
+    sanitized_postcode = UKPostcode.parse(params[:postcode].gsub(/[^\w\s]/i, '').strip).to_s
```

Comment:
It might be nice to extract the sanitization here to an object that can be unit-tested on it's own.  Not that I think what we're doing here has any edge-cases that we've not thought of, but the tests can act as living documentation that the comment may not always do.